database/apikeys.go

```diff
@@ -84,8 +105,37 @@ func (ak *APIKey) LoadBytes(b []byte) error {
 	return nil
 }
 
+// LoadString loads a string into the API key and validates it.
+func (ak *APIKey) LoadString(s string) error {
+	k := APIKey(strings.ToUpper(s))
+	if !k.IsValid() {
+		return ErrInvalidAPIKey
+	}
+	*ak = k
+	return nil
+}
+
+// LoadBytes encodes a []byte of size PubKeySize into an API key.
+func (ak APIKey) String() string {
+	return string(ak)
+}
+
+// CoversSkylink tells us whether a given API key covers a given skylink.
+// Private API keys cover all skylinks while public ones - only a limited set.
+func (akr APIKeyRecord) CoversSkylink(sl string) bool {
+	if !akr.Public {
+		return true
+	}
+	for _, s := range akr.Skylinks {
```

Comment:
what's the limit on `akr.Skylinks` here, this might seem like something we shouldn't consider but if these API keys are going to be evaluated before every single download, shouldn't we consider optimizing this search here?

Reply:
not sure myself to be honest, just thinking out loud, would be nice to find a fast way to tell whether a skylink is covered or not

Reply:
Do I see a real-life use case for bloom filters?! I would love to implement that.